Source/Scene/UrlTemplateImageryProvider.js

```diff
@@ -113,15 +113,22 @@ define([
      *                  be treated as if their alpha is 1.0 everywhere.  When this property is false, memory usage
      *                  and texture upload time are potentially reduced.
      * @param {GetFeatureInfoFormat[]} [options.getFeatureInfoFormats] The formats in which to get feature information at a
-     *                                 specific location when {@see UrlTemplateImageryProvider#pickFeatures} is invoked.  If this
+     *                                 specific location when {@link UrlTemplateImageryProvider#pickFeatures} is invoked.  If this
      *                                 parameter is not specified, feature picking is disabled.
-     *
+     * @param {Boolean} [options.deferReadiness=false] True to defer full construction of this instance until a later call to {@link UrlTemplateImageryProvider#reinitialize}.
```

Comment:
@pjcozzi @mramato I understand how to refactor `UrlTemplateImageryProvider` to allow `options` to be a promise to options, but I'm not entirely sure how to change `createTileMapServiceImageryProvider.js` to create a promise to options and pass that promise into the `UrlTemplateImageryProvider` constructor. 

Please correct me if I have any flaws in this logic:

Currently in `createTileMapServiceImageryProvider`, I initialize a deferred instance of `UrlTemplateImageryProvider` with its `readyPromise` initialized to a deferred object. This allows me to reference that `readyPromise` when the `loadXML` promise in `createTileMapServiceImageryProvider` either resolves or rejects. Once the `readyPromise` resolves, I call `imageryProvider.reinitialize` to pass in the updated `options`.

If the `UrlTemplateImageryProvider` constructor instead took in a promise to options, we would want that promise to resolve to true when the `readyPromise` has resolved to true, as we are populating `options` with data from the `loadXML` promise. However, I can’t reference the `UrlTemplateImageryProvider` instance’s `readyPromise` within the constructor of the imagery provider itself.

I suppose I could create an entirely separate `optionsPromise` or something, but there's still no way to reference the `readyPromise` on [line 169] (https://github.com/tiffanylu/cesium/blob/tileMapServiceImageryProvider/Source/Scene/createTileMapServiceImageryProvider.js#L169) if there is no existing instance of `UrlTemplateImageryProvider`.

Does that make sense, or am I approaching this the wrong way?